marlon/baseline_models/env_wrappers/defend_wrapper.py

```diff
@@ -35,36 +36,38 @@ def __init__(self,
         cyber_env: CyberBattleEnv,
         attacker_reward_store: IRewardStore,
         event_source: Optional[EnvironmentEventSource] = None,
+        defender: bool = False,
         max_timesteps=100,
         enable_action_penalty=True):
         super().__init__()
+        self.defender = None
         self.cyber_env: CyberBattleEnv = cyber_env
         self.bounds: EnvironmentBounds = self.cyber_env._bounds
         self.num_services = 0
         self.observation_space: Space = self.__create_observation_space(cyber_env)
         self.action_space: Space = self.__create_defender_action_space(cyber_env)
-        self.__get_privilegelevel_array = cyber_env._CyberBattleEnv__get_privilegelevel_array
+        self.network_availability: float = 1.0
         self.valid_action_count = 0
         self.invalid_action_count = 0
         self.max_timesteps = max_timesteps
         self.timesteps = 0
         self.rewards = []
         self.attacker_reward_store = attacker_reward_store
-
+        self.first = True
+        self.reset_request = False
         # Add this object as an observer of the cyber env.
         if event_source is None:
             event_source = EnvironmentEventSource()
 
         self.event_source = event_source
         event_source.add_observer(self)
-
-        self.__done = False
+        if defender:
+            self.defender: LearningDefender = LearningDefender(cyber_env)
+        else:
+            logging.error("Attempting to use the defender environment without a defender present.")
```

Comment:
Maybe use an assertion here (assuming bad things happen if you don't see the "error" message)
```suggestion
        assert defender is not None, "Attempting to use the defender environment without a defender present."
        self.defender: LearningDefender = LearningDefender(cyber_env)
```